integration-tests/src/test/java/gov/uk/address/api/stepdefinitions/AddressSteps.java

```diff
@@ -0,0 +1,103 @@
+package gov.uk.address.api.stepdefinitions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jwt.SignedJWT;
+import gov.uk.address.api.client.AddressApiClient;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import uk.gov.di.ipv.cri.common.library.client.ClientConfigurationService;
+import uk.gov.di.ipv.cri.common.library.stepdefinitions.CriTestContext;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class AddressSteps {
+
+    private final ObjectMapper objectMapper;
+    private final AddressApiClient addressApiClient;
+    private final CriTestContext testContext;
+    private String UPRN;
+    private String POSTCODE;
+
+    public AddressSteps(
+            ClientConfigurationService clientConfigurationService, CriTestContext testContext) {
+        this.objectMapper = new ObjectMapper();
+        this.addressApiClient = new AddressApiClient(clientConfigurationService);
+        this.testContext = testContext;
+    }
+
+    @When("the user performs a postcode lookup for post code {string}")
+    public void theUserPerformsAPostcodeLookupForPostCode(String postcode)
+            throws IOException, InterruptedException {
+        this.testContext.setResponse(
+                this.addressApiClient.sendPostCodeLookUpRequest(
+                        this.testContext.getSessionId(), postcode));
+    }
+
+    @Then("user receives a list of addresses containing {string}")
+    public void userReceivesAListOfAddressesContaining(String postcode) throws IOException {
+        JsonNode jsonNode = objectMapper.readTree(this.testContext.getResponse().body());
+        assertEquals(200, this.testContext.getResponse().statusCode());
+        assertNotNull(jsonNode.get(0).get("uprn").asText());
+        UPRN = jsonNode.get(0).get("uprn").asText();
+        assertEquals(postcode, jsonNode.get(0).get("postalCode").asText());
+        POSTCODE = jsonNode.get(0).get("postalCode").asText();
+    }
+
+    @When("the user selects address")
+    public void theUserSelectsAddress() throws IOException, InterruptedException {
+        this.testContext.setResponse(
+                this.addressApiClient.sendAddressRequest(
+                        this.testContext.getSessionId(), UPRN, POSTCODE));
+    }
+
+    @Then("the address is saved successfully")
+    public void theAddressIsSavedSuccessfully() {
+        assertEquals(204, this.testContext.getResponse().statusCode());
+    }
+
+    @When("user sends a POST request to Credential Issue end point with a valid access token")
+    public void user_sends_a_post_request_to_credential_issue_end_point_with_a_valid_access_token()
```

Comment:
remove this underscore naming convention and just write the method name as paer all the others in this class

Reply:
Done.